### ISSUE TITLE
Umlaut in TAP-comments

### ISSUE THREAD

**Issue by mpepa** [suransys/omnistap#13]
I started to use this great framework and it is really fun und works very good.
I ran just into one problem: I made a Jenkins-job to run all my tests and then use 'Publish TAP Results'-step to analyse the tap-files. Jenkins was unable to analyse the files, that had characters like german umlauts in it. I startet to analyse the problem and found out, that the TAP-Files where written as ANSI-files. I changed it to UTF8, but without BOM (very important!) and now everything worked fine.

I changed just these 2 lines in ogFileUtils.$write
Line 35:
```
Do uniconv(kUniTypeUTF32,pcText,pnEncoding,Bin,kFalse,Error)     
Do loFile.$writefile(Bin) Returns lbOK    
```
There is a drawback in this implementation: The functionality of pbAppend is lost.

What do You think about implementing this in the central version? Or should I just fork it?

**Comment by mpepa** [comment]
Unfortunately not, because it will then write a BOM and the Jenkins-TAP-Plugin will not read the TAP-File, not even if it has no Umlaut.

**Comment by barkingfoodog** [comment]
Ah, I see. How about if when `pbAppend` is true, we set the position to the end of the file:
```omnis
24  Do loFile.$openfile(pcPath) Returns lbOK
25  Do loFile.$setposition(loFile.$getfilesize())
```
This would allow use to use `$writefile()` as you suggest and still keep the append functionality.

Would that work?

**Comment by barkingfoodog** [comment]
We could either use `$setposition()` or pass the position as an argument. Per the Omnis docs:

> By default, $writefile() writes the data starting at the current position. You can specify an alternative position using the iStartPos parameter. 

Would you be willing to try this change and submit a PR to add it? If not, I can give it a shot.

And Happy New Year to you as well!

**Comment by mpepa** [comment]
Sorry for my late answer, January has been busy.
I implemented it as You said, but I am not sure, what You mean by 'submit an PR'.
this is the new code in ogFileUtils.$write after Line 36:
```
Do uniconv(kUniTypeUTF32,pcText,pnEncoding,Bin,kFalse,Error)    
If pbAppend   
	Do loFile.$setposition(loFile.$getfilesize())    
End If    
Do loFile.$writefile(Bin) Returns lbOK    
```

**Comment by barkingfoodog** [comment]
My apologies for being cryptic—PR stands for Pull Request. Henk put together a great guide on how to submit changes to an Omnis project on GitHub via a pull request, which is a pretty standard way to contribute to a project: http://omnis.ci/guides/contribute-to-omnis-libraries.html

**Comment by barkingfoodog** [comment]
Fixed in #18 .
